receiver/hostmetricsreceiver/factory.go

```diff
@@ -113,15 +112,15 @@ func createAddScraperOptions(
 	return scraperControllerOptions, nil
 }
 
-func createHostMetricsScraper(ctx context.Context, set receiver.CreateSettings, key string, cfg internal.Config, factories map[string]internal.ScraperFactory) (scraper scraperhelper.Scraper, ok bool, err error) {
+func createHostMetricsScraper(ctx context.Context, set receiver.CreateSettings, key string, cfg internal.Config, factories map[string]internal.ScraperFactory, envMap common.EnvMap) (scraper scraperhelper.Scraper, ok bool, err error) {
```

Comment:
I think we should set the `envMap` to a `internal.Config` field, same way as we are passing `RootPath`, instead of passing it as another argument everywhere.

Reply:
That part is now done.